File: docs/snippets/Serialization/arctrl-json/table-roundtrip.ts
// docs:begin
import {
  ArcInvestigation,
  JsonController,
} from "@nfdi4plants/arctrl";

const investigation = new ArcInvestigation(
  "RoundtripInvestigation",
  "Roundtrip Example",
);

const json = JsonController.Investigation.toJsonString(investigation, 2);
const investigationAgain = JsonController.Investigation.fromJsonString(json);
// docs:end

// docs:assert
if (investigationAgain.Identifier !== "RoundtripInvestigation") {
  throw new Error(
    `Expected investigation identifier to survive JSON roundtrip, got ${investigationAgain.Identifier}`,
  );
}

if (investigationAgain.Title !== "Roundtrip Example") {
  throw new Error("Expected investigation title to survive JSON roundtrip.");
}
// docs:endassert

File: docs/snippets/Serialization/arctrl-json/table-roundtrip.py
# docs:begin
from arctrl import ArcInvestigation, JsonController

investigation = ArcInvestigation(
    "RoundtripInvestigation",
    "Roundtrip Example",
)

json = JsonController.Investigation().to_json_string(investigation, 2)
investigation_again = JsonController.Investigation().from_json_string(json)
# docs:end

# docs:assert
if investigation_again.Identifier != "RoundtripInvestigation":
    raise Exception(
        "Expected investigation identifier to survive JSON roundtrip, "
        f"got {investigation_again.Identifier}"
    )

if investigation_again.Title != "Roundtrip Example":
    raise Exception("Expected investigation title to survive JSON roundtrip.")
# docs:endassert
